### ISSUE TITLE
Non-generic metaclass being flagged as error

### ISSUE THREAD

**Issue by estheruary** [microsoft/pyright#3871]
**Describe the bug**
If you create a metaclass with an `__init__` method pyright thinks it's generic and errs.

**To Reproduce**
Create a metaclass with an init method and use it.

**Expected behavior**
To flag the code as correctly typed.

**Screenshots or Code**
```python
class SomeMetaclass(type):
    def __init__(cls, name, bases, dct):
        pass

class SomeClass(metaclass=SomeMetaclass):
    pass
```

<img width="816" alt="image" src="https://user-images.githubusercontent.com/43628498/187121609-55142cfc-b5dd-49ce-81b4-ec69da686081.png">

If your code relies on symbols that are imported from a third-party library, include the associated import statements and specify which versions of those libraries you have installed.

**VS Code extension or command-line**
```
    "pyright": {
      "version": "1.1.268",
      "resolved": "https://registry.npmjs.org/pyright/-/pyright-1.1.268.tgz",
      "integrity": "sha512-X+8N4BD4mCKFP5pjE5nGSHXz1ysIwCXS6t2Lnxz3kt85mOzh9Q2GNBNPYEW4svuRjfJm/8f7MIc4jyHFjsJwSw=="
    },
```

**Additional context**
Maybe related to https://github.com/microsoft/pyright/issues/2683

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

When pyright encounters a class with an unannotated `__init__` method, it treats the class as generic under the covers, acting as though it has a different type variable for each input parameter. That allows for better inference of instance variables, which are often initialized from the constructor's input parameters. For example:

```python
class Foo:
    def __init__(self, a, b):
        self.a = a
        self.b = b

f = Foo(1, "hi")

reveal_type(f.a) # int
reveal_type(f.b) # str
```

Internally, we refer to these as "pseudo-generic" classes because they're not explicitly declared as generic.

In this particular case, we generated a false positive error because of this.

If you don't want to wait for the next release of pyright, you can work around the problem by adding annotations to one or more of the input parameters for the `__init__` method.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.269, which I just published. It will also be included in a future version of pylance.
